microbench/db_basic_bench.cc

```diff
@@ -129,6 +246,1019 @@ static void DBClose(benchmark::State& state) {
+      }
+    }
+  }
+  b->ArgNames({"comp_style", "max_data", "per_key_size"});
+}
+
+static const uint64_t IteratorPrevNum = 10l << 10;
+BENCHMARK(IteratorPrev)
+    ->Iterations(IteratorPrevNum)
+    ->Apply(IteratorPrevArguments);
+
+static void PrefixSeek(benchmark::State& state) {
+  auto compaction_style = static_cast<CompactionStyle>(state.range(0));
+  uint64_t max_data = state.range(1);
+  uint64_t per_key_size = state.range(2);
+  bool enable_statistics = state.range(3);
+  bool enable_filter = state.range(4);
+  uint64_t key_num = max_data / per_key_size;
+
+  // setup DB
+  static DB* db;
+  Options options;
+  if (enable_statistics) {
+    options.statistics = CreateDBStatistics();
+  }
+  options.compaction_style = compaction_style;
+  options.prefix_extractor.reset(NewFixedPrefixTransform(4));
+
+  if (enable_filter) {
+    BlockBasedTableOptions table_options;
+    table_options.filter_policy.reset(NewBloomFilterPolicy(10, false));
+    options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+  }
+
+  auto rnd = Random(301 + state.thread_index);
+  KeyGenerator kg(&rnd, key_num, key_num / 100);
+
+  if (state.thread_index == 0) {
+    SetupDB(state, options, &db, "PrefixSeek");
+
+    // load db
+    auto wo = WriteOptions();
+    wo.disableWAL = true;
+    for (uint64_t i = 0; i < key_num; i++) {
+      auto s = db->Put(wo, kg.Next(), rnd.RandomString(per_key_size));
+      if (!s.ok()) {
+        state.SkipWithError(s.ToString().c_str());
+      }
+    }
+
+    FlushOptions fo;
+    auto s = db->Flush(fo);
+    if (!s.ok()) {
+      state.SkipWithError(s.ToString().c_str());
+    }
+
+    auto db_full = static_cast_with_check<DBImpl>(db);
+    s = db_full->WaitForCompact(true);
+    if (!s.ok()) {
+      state.SkipWithError(s.ToString().c_str());
+      return;
+    }
+  }
+
+  {
+    auto ro = ReadOptions();
+    size_t not_found = 0;
+    std::unique_ptr<Iterator> iter(db->NewIterator(ro));
+    for (auto _ : state) {
+      iter->Seek(kg.NextPrefix());
+      if (!iter->status().ok()) {
+        state.SkipWithError(iter->status().ToString().c_str());
+        return;
+      }
+    }
+  }
+
+  if (state.thread_index == 0) {
+    TeardownDB(state, db, options, kg);
+  }
+}
+
+static void PrefixSeekArguments(benchmark::internal::Benchmark* b) {
+  for (int comp_style : {kCompactionStyleLevel, kCompactionStyleUniversal,
+                         kCompactionStyleFIFO}) {
+    for (int64_t max_data : {128l << 20, 512l << 20}) {
+      for (int64_t per_key_size : {256, 1024}) {
+        for (bool enable_statistics : {false, true}) {
+          for (bool enable_filter : {false, true}) {
+            b->Args({comp_style, max_data, per_key_size, enable_statistics,
+                     enable_filter});
+          }
+        }
+      }
+    }
+  }
+  b->ArgNames({"comp_style", "max_data", "per_key_size", "enable_statistics",
+               "enable_filter"});
+}
+
+static const uint64_t PrefixSeekNum = 10l << 10;
```

Comment:
Nit: constexpr may be better, but this is minor. The name should be `kPrefixSeekNum`.